src/components/DocExplorer.js

```diff
@@ -148,6 +176,111 @@ export class DocExplorer extends React.Component {
   }
 }
 
+// Render Search Results
+class SearchDoc extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    return (this.props.schema !== nextProps.schema) ||
+      (this.props.searchValue !== nextProps.searchValue);
+  }
+
+  render() {
+    const searchValue = this.props.searchValue;
+    const schema = this.props.schema;
+    const onClickType = this.props.onClickType;
+    const onClickField = this.props.onClickField;
+
+    const typeMap = schema.getTypeMap();
+
+    let matchedTypes = [];
+    let matchedFields = [];
+    let matchedArgs = [];
+
+    Object.keys(typeMap).forEach(typeName => {
+      const type = typeMap[typeName];
+      let matchedOn = [];
+      if (typeName.indexOf(searchValue) !== -1) {
+        matchedOn.push('Type Name');
+      }
+
+      if (matchedOn.length) {
+        matchedTypes.push(
+          <div className="doc-category-item">
+            <TypeLink type={type} onClick={onClickType} />
+          </div>
+        );
+      }
+
+      if (type.getFields) {
+        let fields = type.getFields();
+        Object.keys(fields).forEach(fieldName => {
+          let field = fields[fieldName];
+          if (fieldName.indexOf(searchValue) !== -1) {
+            matchedFields.push(
+              <div className="doc-category-item">
+                <a className="field-name"
+                  onClick={event => onClickField(field, type, event)}>
+                  {field.name}
+                </a>
+                {' on '}
+                <TypeLink type={type} onClick={onClickType} />
+              </div>
+            );
+          } else if (field.args && field.args.length) {
+            let matches = field.args.filter(arg => {
+              return arg.name.indexOf(searchValue) !== -1;
```

Comment:
you can make this a directly returned expression instead of a function block

```
const matches = field.args.filter(arg => arg.name.indexOf(searchValue) !== -1);
```